### PR TITLE
log-backup: Refactor daemon

### LINKED ISSUE #36762: log-backup: elide listening to tasks when the node isn't owner
## Enhancement

For now, we start listen task change for each TiDB nodes embedding the advancer.

We can do some refactor for making only the owner creates the listen stream.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #36762 

Problem Summary:

See the issue.

### What is changed and how it works?

The `AdvancerDaemon` has been refactored, now, it would be carried by a new type `OwnerDaemon`, which supports the callback `OnBegin`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: br/pkg/streamhelper/daemon/interface.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
+
+package daemon
+
+import "context"
+
+// Interface describes the lifetime hook of a daemon application.
+type Interface interface {
```
Comment (3pointer) [inline br/pkg/streamhelper/daemon/interface.go:8]: can we have a better name?

Comment (YuJuncen) [reply br/pkg/streamhelper/daemon/interface.go:8]: This name style is ported from [`sort.Interface`](https://pkg.go.dev/sort@go1.19.1#Interface), in some way, I guess using the name `Interface` to name the main interface of the package is pretty go-style. 🤔

Comment (3pointer) [reply br/pkg/streamhelper/daemon/interface.go:8]: OK

**Review by 3pointer — APPROVED** [review]

**Pull request merged**